server/bin/pbench-server-prep-shim-002.py

```diff
@@ -0,0 +1,331 @@
+        ntotal += 1
+
+        # full pathname of tarball
+        tb = Path(tbmd5[0:-4])
+        tbmd5 = Path(tbmd5)
+
+        # directory
+        tbdir = tb.parent
+
+        # resultname: get the basename foo.tar.xz and then strip the .tar.xz
+        resultname = tb.name
+
+        controller = tbdir.name
+        dest = archive / controller
+
+        if (
+            (dest / resultname).is_file()
+            and (dest / tbmd5.name).is_file()
+            and (dest / resultname).exists()
+            and (dest / tbmd5.name).exists()
+        ):
+            ef.write(f"{config.TS}: Duplicate: {tb} duplicate name\n")
+            quarantine((duplicates / controller), logger, tb, tbmd5)
+            ndups += 1
+            continue
+
+        archive_tar_hex_value, archive_md5_hex_value = md5_check(tb, tbmd5, logger)
+        if archive_tar_hex_value != archive_md5_hex_value:
+            ef.write(f"{config.TS}: Quarantined: {tb} failed MD5 check\n")
+            logger.info("{}: FAILED", tb.name)
+            logger.info("md5sum: WARNING: 1 computed checksum did NOT match")
+            quarantine((qdir_md5 / controller), logger, tb, tbmd5)
+            nquarantined += 1
+            continue
+
+        # make the destination directory and its TODO subdir if necessary.
+        try:
+            os.makedirs(dest / "TODO")
+        except FileExistsError:
+            # directory already exists, ignore
+            pass
+        except Exception:
+            logger.error("{}: Error in creating TODO directory.", config.TS)
+            quarantine(os.path.join(errors, controller), logger, tb, tbmd5)
+            nerrs += 1
+            continue
+
+        # First, copy the small .md5 file to the destination. That way, if
+        # that operation fails it will fail quickly since the file is small.
+        try:
+            shutil.copy2(tbmd5, dest)
+        except Exception:
+            logger.error(
+                "{}: Error in copying .md5 file to Destination path.", config.TS
+            )
+            try:
+                os.remove(dest / tbmd5.name)
+            except Exception:
+                logger.error(
+                    "{}: Warning: cleanup of copy failure failed itself.", config.TS
+                )
+            quarantine((errors / controller), logger, tb, tbmd5)
+            nerrs += 1
+            continue
+
+        # Next, mv the "large" tar ball to the destination. If the destination
+        # is on the same device, the move should be quick. If the destination is
+        # on a different device, the move will be a copy and delete, and will
+        # take a bit longer.  If it fails, the file will NOT be at the
+        # destination.
+        try:
+            shutil.move(str(tb), str(dest))
+        except Exception:
+            logger.error(
+                "{}: Error in moving tarball file to Destination path.", config.TS
+            )
+            try:
+                os.remove(dest / resultname)
```

Comment:
Same comment as above, for failure conditions, the destination tar ball may not exist, so we should only log this exception if it is not "file not found".

Reply:
ok